### PR TITLE
fix(ext/http): websocket upgrade header check

### LINKED ISSUE #11747: Lowercase on websocket upgrade header check
I have a websocket server with deno 1.13 and it started giving the following error when connecting:

```
Invalid Header: 'upgrade' header must be 'websocket'
```

I saw that the update check is case sensitive:

https://github.com/denoland/deno/blob/4b800f5d269de259da29d3b42dae66faf4351476/ext/http/01_http.js#L320

I know the default is 'websocket', but some clients are third-party and it's hard to change their codes.

Is It possible considering a upgrade header with no case sensitive?

### PR DESCRIPTION
**Pull request by galiazzi**
Fixes #11747

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Thanks for the PR @galiazzi, any chance we could get a test exercising this change?

File: ext/http/01_http.js [hunk 1]
```diff
@@ -341,7 +341,8 @@
   const _ws = Symbol("[[associated_ws]]");
 
   function upgradeWebSocket(request, options = {}) {
-    if (request.headers.get("upgrade") !== "websocket") {
+    const upgrade = request.headers.get("upgrade");
+    if (!upgrade || upgrade.toLowerCase() !== "websocket") {
```
Comment (lucacasonato) [inline ext/http/01_http.js:345]: Please use "primordials" to do this check. You can import `StringPrototypeToLowerCase` (see top of file how), and then do `StringPrototypeToLowerCase(upgrade) !== "websocket"`.

**Review by lucacasonato — COMMENTED** [review]
Looks good, I just have one small change request.

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @galiazzi

**Pull request merged**